test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2080,6 +2193,352 @@ def test_cgroup_use_hierarchy(self):
         else:
             self.assertFalse(1, "File %s not present" % fpath)
 
+    def check_req_rjs(self):
+        """
+        Check the requirements for the reliable job startup tests.
+        Return 1 if requirements are not satisfied.
+        """
+        ret = 0
+        # Check that momA has at least two free vnodes
+        attr = {'state': 'free'}
+        rv1 = True
+        try:
+            self.server.expect(VNODE, attr, id='%s[0]' % self.hostA,
+                               max_attempts=3, interval=2)
+        except PtlExpectError, e:
+            rv1 = e.rv
+
+        rv2 = True
+        try:
+            self.server.expect(VNODE, attr, id='%s[1]' % self.hostA,
+                               max_attempts=3, interval=2)
+        except PtlExpectError, e:
+            rv2 = e.rv
+
+        # Check that momB has at least one free vnode
+        rv3 = True
+        try:
+            self.server.expect(VNODE, attr, id='%s[0]' % self.hostB,
+                               max_attempts=3, interval=2)
+        except PtlExpectError, e:
+            rv3 = e.rv
+
+        if not rv1 or not rv2 or not rv3:
+            ret = 1
+        return ret
+
+    def test_cgroup_release_nodes(self):
+        """
+        Verify that exec_vnode values are trimmed
+        when execjob_launch hook prunes job via release_nodes(),
+        tolerate_node_failures=job_start
+        """
+        # Test requires 2 nodes
+        if len(self.moms) != 2:
+            self.skipTest('Test requires two Moms as input, '
+                          'use -p moms=<mom1:mom2>')
+        self.load_config(self.cfg7)
+
+        # Check that moms have the required free vnodes
+        if self.check_req_rjs() == 1:
+            self.skipTest('There are not enough free vnodes to run the test')
```

Comment:
There may be enough if hostB has two and hostA has one, but the check will fail in that case. I think you should explicitly state that hostA must have more than one vnode while hostB only needs one.

Reply:
I have changed the skipTest comment to be more explicit.